pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/RowDataBlock.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.blocks;
```

Comment:
high level question - If a query is planned by the new multi-stage planner in way that there are 2 stages (leaf servers and broker) like today and there is no exchange needed, then the TransferableBlock will not be used at all right ? In other words, we will continue to leverage DataTable for such cases or do we plan on using BaseDataBlock throughout ?

Reply:
so long story short: we don't have reduce stage at all (exception select-only-without-ordering reduce which is just a concat), so we need to migrate the reduce stage code (agg / distinct / orderby.... to the intermediate stage (worker operator))